cmd/soroban-cli/src/utils.rs

```diff
@@ -196,6 +196,26 @@ pub mod http {
     }
 }
 
+pub mod args {
+    #[derive(thiserror::Error, Debug)]
+    pub enum DeprecatedError<'a> {
+        #[error("This argument has been removed and will be not be recognized by the future versions of CLI: {0}"
+        )]
+        RemovedArgument(&'a str),
+    }
+
+    #[macro_export]
+    macro_rules! removed_arg {
```

Comment:
I think we should make this function a little more explicit as to it's intended effect, so it's clear it isn't going to pass the value, but it's a way to display a meaningful message and then error.
```suggestion
    macro_rules! error_on_use_of_removed_arg {
```